gradle-jdks-setup/src/main/resources/gradle-jdks-functions.sh

```diff
@@ -0,0 +1,156 @@
+#
+# (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+TMP_WORK_DIR=$(mktemp -d)
+export TMP_WORK_DIR
+
+cleanup() {
+  [ -d "$TMP_WORK_DIR" ] && rm -rf "$TMP_WORK_DIR"
+}
+
+die() {
+    echo
+    echo "$*"
+    echo
+    cleanup
+    exit 1
+} >&2
+
+read_value() {
+  if [ ! -f "$1" ]; then
+    die "ERROR: $1 not found, aborting Gradle JDK setup"
+  fi
+  read -r value < "$1" || die "ERROR: Unable to read value from $1. Make sure the file ends with a newline."
+  echo "$value"
+}
+
+get_os() {
+  # OS specific support; same as gradle-jdks:com.palantir.gradle.jdks.setup.common.CurrentOs.java
+  case "$( uname )" in                          #(
+    Linux* )          os_name="linux"  ;;       #(
```

Comment:
This actually makes a global variable called `os_name`. If any other functions use os_name as variable it would get clobbered. Probably fine as bash is all single threaded, but best to declare this at the top as `local os_name` so it's scoped only to this function.

Reply:
Same for all the other function scoped variables here.

Reply:
Ah, oh well. I guess `local` is bash-ism. I feel even with the underscore at the front these might clash between functions. We have tests to make sure we haven't messed it up, so it's probably fine.